inc/zoo/swar/SWAR.h

```diff
@@ -68,7 +78,26 @@ struct SWAR {
                 ~(~T(0) << NBits),
         // Use LowerBits in favor of ~MostSignificantBit to not pollute
         // "don't care" bits when non-power-of-two bit lane sizes are supported
-        LowerBits = MostSignificantBit - LeastSignificantBit;
+        LowerBits = MostSignificantBit - LeastSignificantBit,
+        MaxUnsignedLaneValue = LeastSignificantLaneMask;
+
+    template <typename U, typename ManipulationFn>
+    constexpr auto loadBaseTypeIntoLanes(const U (&values)[Lanes],
+                                         const ManipulationFn&& manipulation) {
+        auto result = T{0};
+        for (auto value : values) {
+            auto laneValue = manipulation(value);
+            result = (result << NBits) | laneValue;
+        }
+        return result;
+    }
```

Comment:
not 100% sure about the names for `manipulation` etc right now but would be good to get your opinion on this approach.

Reply:
This might be a reasonable way to do this, but I'd use the appropriate algorithm in `#include <algorithm>` and I would not take by `const`-r-value reference something that ought to be a pure forwarding reference, meaning `ManipulationFN &&`
Make sure you never put `const`-r-value, either `const`-l-value when strictly necessary, or forwarding reference whenever possible.